### PR TITLE
List accessor

### LINKED ISSUE #55323: ENH: Add list accessor for PyArrow list types to support __getitem__
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pyarrow as pa

# works
pd.Series(["tsht(thsat","gda(gr"]).str.split("(").str[0]
# works
pd.Series(["tsht(thsat","gda(gr"], dtype="string[pyarrow]").str.split("(").str[0]
# fails
pd.Series(["tsht(thsat","gda(gr"], dtype=pd.ArrowDtype(pa.string())).str.split("(").str[0]
```

### Issue Description

    raise AttributeError("Can only use .str accessor with string values!")
AttributeError: Can only use .str accessor with string values!

### Expected Behavior

pd.Series(["tsht(thsat","gda(gr"], dtype=pd.ArrowDtype(pa.string())).str.split("(").str[0]

should do the same as the rest, also i would not use pd.ArrowDtype(pa.string(), but it just happens to be the default datatype when reading a file using pd.read_excel.

### Installed Versions

<details>

DEBUG - matplotlib: matplotlib data path: C:\Users\PabloRuiz\AppData\Local\pypoetry\Cache\virtualenvs\transformer-wi-nHELc-py3.11\Lib\site-packages\matplotlib\mpl-data
DEBUG - matplotlib: CONFIGDIR=C:\Users\PabloRuiz\.matplotlib
DEBUG - matplotlib: interactive is False
DEBUG - matplotlib: platform is win32
INSTALLED VERSIONS
------------------
commit              : e86ed377639948c64c429059127bcf5b359ab6be
python              : 3.11.1.final.0
python-bits         : 64
OS                  : Windows
OS-release          : 10
Version             : 10.0.19045
machine             : AMD64
processor           : Intel64 Family 6 Model 140 Stepping 1, GenuineIntel
byteorder           : little
LC_ALL              : None
LANG                : None
LOCALE              : English_United States.1252
pandas              : 2.1.1
numpy               : 1.26.0
pytz                : 2023.3.post1
dateutil            : 2.8.2
setuptools          : 68.2.2
pip                 : 22.3.1
Cython              : None
pytest              : 7.4.2
hypothesis          : None
sphinx              : None
blosc               : None
feather             : None
xlsxwriter          : 3.1.4
lxml.etree          : None
html5lib            : None
pymysql             : None
psycopg2            : None
jinja2              : 3.1.2
IPython             : 8.15.0
pandas_datareader   : None
bs4                 : None
bottleneck          : None
dataframe-api-compat: None
fastparquet         : None
fsspec              : None
gcsfs               : None
matplotlib          : 3.8.0
numba               : None
numexpr             : None
odfpy               : None
openpyxl            : 3.1.3
pandas_gbq          : None
pyarrow             : 13.0.0
pyreadstat          : None
pyxlsb              : 1.0.10
s3fs                : None
scipy               : None
sqlalchemy          : 1.4.31
tables              : None
tabulate            : None
xarray              : None
xlrd                : 2.0.1
zstandard           : None
tzdata              : 2023.3
qtpy                : None
pyqt5               : None

</details>

### PR DESCRIPTION
**Pull request by rohanjain101**
- [ ] closes #55323(Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/arrow/accessors.py [hunk 1]
```diff
@@ -19,7 +23,182 @@
...
+        0    3
+        1    1
+        dtype: int32[pyarrow]
+        """
+        from pandas import Series
+
+        value_lengths = pc.list_value_length(self._pa_array)
+        return Series(value_lengths, dtype=ArrowDtype(value_lengths.type))
+
+    def __getitem__(self, key: int | slice) -> Series:
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/accessors.py:111]: Could you also define `__iter__` and make it raise a TypeError? You can reference `StringMethods.__iter__`

**Review by mroeschke — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/arrow/accessors.py (hunk 1 above)
Comment (rohanjain101) [reply pandas/core/arrays/arrow/accessors.py:113]: Done.

**Comment by rohanjain101** [comment]
Not sure why doc string test is failing.

File: pandas/core/arrays/arrow/accessors.py [hunk 2]
```diff
@@ -13,13 +20,194 @@
...
+    from collections.abc import Iterator
+
     from pandas import (
         DataFrame,
         Series,
     )
 
 
-class StructAccessor:
+class _ArrowAccessor(metaclass=ABCMeta):
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/accessors.py:31]:
```suggestion
class ArrowAccessor(metaclass=ABCMeta):
```

Nit: No need to make this private

**Review by mroeschke — COMMENTED** [review]
Minor comment and some mypy issues otherwise LGTM

**Comment by rohanjain101** [comment]
@mroeschke I'm not really sure how to work around the CI failure, it looks like the issue is that I'm calling _pa_array in arrow accessor which is only defined in the ArrowExtensionArray, but it's also impossible for the Series to not be backed by an ArrowExtensionArray because of the call to _validate in init.

I looked at the old code in StructAccessor and it was also accessing _pa_array directly, not sure how that one passed the check and this one doesn't. Would appreciate some guidance with this.

File: pandas/core/arrays/arrow/accessors.py (hunk 2 above)
Comment (rohanjain101) [reply pandas/core/arrays/arrow/accessors.py]: Done, thanks,

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Nice work @rohanjain101!

**Pull request merged**